qiskit/mapper/_layout.py

```diff
@@ -48,62 +50,70 @@ def from_list(self, input_list):
             [(QuantumRegister(3, 'qr'), 0), None,
              (QuantumRegister(3, 'qr'), 2), (QuantumRegister(3, 'qr'), 3)]
         """
-        for key, value in enumerate(input_list):
-            self[key] = value
+        for physical, virtual in enumerate(input_list):
+            if Layout.is_virtual(virtual):
```

Comment:
Maybe raise an exception if virtual is invalid, or is there a case where we would want to drop it?

Reply:
good one. Done.